RFC/src/RFC-0132_Merge_Mining_Monero.md

```diff
@@ -0,0 +1,172 @@
+
+A new Tari block template is obtained from a Tari base node by calling the `get_new_block_template` GRPC method, setting `Monero` as the chosen PoW algorithm.
+The `Monero` algorithm must be selected so that the correct mining difficulty for the Monero algorithm is returned. Remember, that Monero and SHA difficulties 
+are independent (See [RFC-0131_Mining.html](/RFC-0131_Mining.html)).
+
+Next, construct a _Blake256_ hash of _some_ of the Tari header fields. We'll call this hash the merge mining hash \\( h_m \\) that commits to 
+the following header fields in order: `version`, `height`,`prev_hash`,`timestamp`,`output_mr`,`range_proof_mr`,`output_mmr_size`,`kernel_mr`,
+`kernel_mmr_size`,`total_kernel_offset`,`total_script_offset`. Note, this hash does not include the `pow` and `nonce` fields, as these fields are set as part of mining.
+
+To have the chance of mining a Monero block as well as a Tari block, we must obtain a new valid monero block template, by calling [get_block_template].
+This returns a `blocktemplate_blob`, that is, a serialized Monero block containing the Monero block header, coinbase and a list of hashes referencing the 
+transactions included in the block. Additionally, a `blockhashing_blob` is a fixed size blob containing `serialized_monero_header`, `merkle_tree_root` and 
+`txn_count` concatenated together. The `merkle_tree_root` is a _merkle root_ of the coinbase + the transaction hashes contained in the block.
+
+```rust,ignore
+pub struct Block {
+    /// The block header
+    pub header: BlockHeader,
+    /// Coinbase transaction a.k.a miner transaction
+    pub miner_tx: Transaction,
+    /// References to the transactions included in this block
+    pub tx_hashes: Vec<hash::Hash>,
+}
+```
+_fig 1. The Monero block struct_
+
+Next, modify the Monero block template by including the merge mining hash \\( h_m \\) in the extra fields of the coinbase transaction. Monero has a [merge mining subfield] 
+to accommodate this data. Importantly, the extra field data part of the coinbase transaction hash and therefore the `merkle_tree_root`, the `blockhashing_blob` must be 
+reconstructed. A rust port of [Monero's tree hash] algorithm is needed to achieve this. The coinbase hash MUST be the first element to be hashed when constructing the `merkle_tree_root`.
+This satisfies **REQ 2**, proving that the Tari block was constructed before the parent chain's block. 
+
+The block may now be mined. Once a solution is found that satisfies the Tari difficulty, the miner must include enough data to allow the Tari blockchain to assert **REQ 1** and **REQ 2**.
+Concretely, A miner must serialize `MoneroPowData` using Monero consensus encoding and add it to the `pow_data` field in the Tari header.
+
+```rust,ignore
+pub struct MoneroPowData {
+    /// Monero header fields
+    header: MoneroBlockHeader,
+    /// randomX vm key
+    randomx_key: FixedByteArray, // Fixed 64 bytes
+    /// transaction count
+    transaction_count: u16,
```

Comment:
Might be more compact to serialize this with VarInt's

Reply:
Will add some more details on the serialization and mention the VarInt encoding for all ints

Reply:
Added a Serialisation section